### ISSUE TITLE
Resize delay when toggling maximize or full screen

### ISSUE THREAD

**Issue by CendioOssman** [TigerVNC/tigervnc#1763]
**Describe the bug**
When toggling maximize or full-screen mode, there is an annoying delay before the session resizes to the new client size.

**To Reproduce**
1. Connect in windowed mode
2. Maximize client, or enable full screen

**Expected behavior**
Session is immediately the new size

**Client (please complete the following information):**
 - OS: All
 - VNC client: TigerVNC
 - VNC client version: fb7b956cf6c6cf2d2550c25a7112999983dea96e
 - Client downloaded from: Own build

**Server (please complete the following information):**
 - OS: Linux
 - VNC server: TigerVNC
 - VNC server version: fb7b956cf6c6cf2d2550c25a7112999983dea96e
 - Server downloaded from: Own build
 - Server was started using: `Xvnc`

**Additional context**
The current behaviour is partly by design. We want a delay when the user is interactively resizing the client, to avoid a lot of unnecessary resizing of the session. But maximize and full screen are more stable, so the delay is not needed there.

**Comment by hildred** [comment]
> But maximize and full screen are more stable, so the delay is not needed there.


Not exactly. On windows ten when you drag a window to the screen border, it maximizes. (which I find annoying.) I do not know what the right thing to do is.

**Comment by CendioOssman** [comment]
Ah, indeed. That happens on most Linux systems as well. Still, I think the delay is probably unnecessary even there as there is some resistance to toggling in and out of that state.
